### PR TITLE
Add Corretto, FreeFem, Ghostscript, TeXstudio, Scratch and 4 more

### PR DESCRIPTION
**Pull request by adotcoop**
## Added 9 apps and minor modification to Get-Architecture.ps1

- Sourceforge based: gretl, SAGA-GIS
- Github based: FreeFem, Gephi, Ghostscript, Praat, TeXstudio
- Download redirects: Amazon Corretto JDK, Scratch

### Sourceforge apps

- SAGA-GIS
- gretl

gretl version numbers are in the format _year_ _letter_, such as 2020a or 2021b. This required a different regex for MatchVersion "((\\d+.){1})"

### Github apps

- FreeFem
- Gephi
- Ghostscript
- Praat
- TeXstudio

For Ghostscript $VersionTag = "tag_name" did not report the correct version, but "name" did, so to obtain the correct version required the use of "VersionTag": "name"

### Direct download with redirect

- Amazon Corretto JDK
- MIT Scratch

Amazon Corretto is a bit more complicated because they offer multiple different JDK versions and, depending on version, there may be zip downloads or JRE downloads. The pages detailing the permanent URLs are as follows

- https://docs.aws.amazon.com/corretto/latest/corretto-8-ug/downloads-list.html
- https://docs.aws.amazon.com/corretto/latest/corretto-11-ug/downloads-list.html
- https://docs.aws.amazon.com/corretto/latest/corretto-15-ug/downloads-list.html
- https://docs.aws.amazon.com/corretto/latest/corretto-16-ug/downloads-list.html

### Get-Architecture.ps1

To support ghostscript (gs9540w32.exe and gs9540w64.exe), had to add the following lines to Get-Architecture

        "w64" { $architecture = "x64"; Break }
        "w32" { $architecture = "x86"; Break }

To support FreeFem (FreeFEM-4.9-win7-64.exe) and gretl (gretl-2021b-32.exe and gretl-2021b-64.exe), added-

        "-64" { $architecture = "x64"; Break }
        "-32" { $architecture = "x86"; Break }
I was tempted to leave out the 32-bit detections as the script defaults to 32-bit anyway, but left them in for completeness.

### CODE REVIEW

File: Evergreen/Apps/Get-Ghostscript.ps1 [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Function Get-Ghostscript {
```
Comment (aaronparker) [inline Evergreen/Apps/Get-Ghostscript.ps1:1]: Update with vendor name "Artifex", i.e., "ArtifexGhostscript"

File: Evergreen/Manifests/Praat.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+	"Name": "Praat",
```
Comment (aaronparker) [inline Evergreen/Manifests/Praat.json:2]: Is the vendor "Phonetic Sciences"?

File: Evergreen/Manifests/AmazonCorretto.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	"Source": "https://aws.amazon.com/corretto/",
+	"Get": {
+		"Update": {
+			"Uri": ""
+		},
+		"Download": {
+			"JDK": {
+				"8": {
+					"x64msi" :"https://corretto.aws/downloads/latest/amazon-corretto-8-x64-windows-jdk.msi",
+					"x64zip" :"https://corretto.aws/downloads/latest/amazon-corretto-8-x64-windows-jdk.zip",
```
Comment (aaronparker) [inline Evergreen/Manifests/AmazonCorretto.json:12]: Are the zip files used for deployments? I don't know enough about the JDK deployment to answer this one myself. If used, leave them in.

**Review by aaronparker — CHANGES_REQUESTED** [review]
Nice work. Some minor comments before merging.

File: Evergreen/Manifests/AmazonCorretto.json (hunk 3 above)
Comment (adotcoop) [reply Evergreen/Manifests/AmazonCorretto.json:12]: You're probably right - I don't use them myself and they cause 6 additional invoke-webrequests. I've removed them to speed up the function

File: Evergreen/Apps/Get-Ghostscript.ps1 (hunk 1 above)
Comment (adotcoop) [reply Evergreen/Apps/Get-Ghostscript.ps1]: Updated to match Vendor.ApplicationName

File: Evergreen/Manifests/Praat.json (hunk 2 above)
Comment (adotcoop) [reply Evergreen/Manifests/Praat.json:2]: There's no entry in winget for this app, but the chocolatey entry lists the authors as Paul Boersma and David Weenink. This is one of those apps that is widely used in academic environments and written by fellow academics. The authors work in the Phonetic Sciences department of the University of Amsterdam - unsure as to what vendor to put here. I've left it as is just now, but can change if required

- https://community.chocolatey.org/packages/praat

**Pull request merged**